x/evm/types/msg.go

```diff
@@ -295,13 +298,26 @@ func (msg MsgEthereumTx) GetEffectiveFee(baseFee *big.Int) *big.Int {
 }
 
 // GetFrom loads the ethereum sender address from the sigcache and returns an
-// sdk.AccAddress from its bytes
+// sdk.AccAddress from its bytes. If the sender is not found, it returns nil.
+// If the message has an OnBehalf field, it returns the address from that field.
 func (msg *MsgEthereumTx) GetFrom() sdk.AccAddress {
```

Comment:
I think should rename `GetFrom` method or create another with name reflect the logic like actual name or smt because this really confused when reading.

Reply:
sure, renamed it to `GetEffectiveSender`. hope it makes more sense!